misc/multistream-select/src/protocol/dialer.rs

```diff
@@ -44,7 +44,7 @@ where
     R: AsyncRead + AsyncWrite,
     N: AsRef<[u8]>
 {
-    pub fn new(inner: R) -> DialerFuture<R, N> {
+    pub fn dial(inner: R) -> DialerFuture<R, N> {
```

Comment:
What did clippy complain about for this?
I'm not a fan of breaking the API just for fun.

Reply:
https://rust-lang.github.io/rust-clippy/master/index.html#new_ret_no_self

Should `multistream_select::protocol` even be a public module?

Reply:
> Should multistream_select::protocol even be a public module?

Not *necessarily*, it's public "just in case".

Reply:
I propose to make it private unless someone asks for it. It is certainly not used outside of multistream-select within libp2p.

Reply:
Well, ok. Let's make it private.